hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java

```diff
@@ -907,6 +906,20 @@ public boolean isServerOnline(ServerName serverName) {
     return serverName != null && onlineServers.containsKey(serverName);
   }
 
+  public enum ServerLiveState {
+    LIVE,
+    DEAD,
+    UNKNOWN
+  }
+
+  /**
+   * @return whether the server is online, dead, or unknown.
+   */
+  public synchronized ServerLiveState isServerKnownAndOnline(ServerName serverName) {
```

Comment:
Do we really need synchronized here? The onlineServers is already a ConcurrentMap.

Reply:
It's needed to synchronize between online and dead map, in case the server moves from one to the other so we miss it from both

Reply:
Oh, we also checked the deadServers. That's fine.